sensor_interface/CMakeLists.txt

```diff
@@ -5,18 +5,25 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-add_subdirectory(src)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wpedantic -Werror -Wextra")
 IF(CMAKE_BUILD_TYPE MATCHES Release)
 	message("Building project Sensor Interface ${VERSION}")
 	message(STATUS "CXX Flags used: ${CMAKE_CXX_FLAGS_RELEASE}")
+	add_subdirectory(src)
 ELSEIF(CMAKE_BUILD_TYPE MATCHES Debug)
 	message("Building project Sensor Interface with tests")
-	message(STATUS "CXX Flags used: ${CMAKE_CXX_FLAGS_DEBUG}")
+	set(CMAKE_CXX_FLAGS "-g -fprofile-arcs -ftest-coverage")
+	add_subdirectory(src)
 	enable_testing()
 	add_subdirectory(test/googletest)
 	add_subdirectory(test/unit_tests)
 	IF(COVERAGE)
+	include("CMakeModules/CodeCoverage.cmake")
+	setup_target_for_coverage_gcovr_html(NAME coverage
+							           	EXECUTABLE ctest
+										EXCLUDE "${PROJECT_SOURCE_DIR}/test/googletest"
+								        )
```

Comment:
please exclude unit test, they are not part of coverage report

Reply:
done